mysql/datadog_checks/mysql/activity.py

```diff
@@ -0,0 +1,292 @@
+        except Exception as e:
+            self._log.error("Failed to collect activity | err=[%s]", e)
+        return []
+
+    def _normalize_rows(self, rows):
+        # type: (List[Dict[str]]) -> List[Dict[str]]
+        rows = sorted(rows, key=lambda r: self._sort_key(r))
+        normalized_rows = []
+        estimated_size = 0
+        for row in rows:
+            row = self._obfuscate_and_sanitize_row(row)
+            estimated_size += self._get_estimated_row_size_bytes(row)
+            if estimated_size > MySQLActivity.MAX_PAYLOAD_BYTES:
+                return normalized_rows
+            normalized_rows.append(row)
+        return normalized_rows
+
+    @staticmethod
+    def _sort_key(row):
+        # type: (Dict[str]) -> int
+        # value is in picoseconds
+        return row.get('event_timer_start') or int(round(time.time() * 1e12))
+
+    def _obfuscate_and_sanitize_row(self, row):
+        # type: (Dict[str]) -> Dict[str]
+        row = self._sanitize_row(row)
+        if "sql_text" not in row:
+            return row
+        try:
+            self._finalize_row(row, obfuscate_sql_with_metadata(row["sql_text"], self._obfuscator_options))
+        except Exception as e:
+            row["sql_text"] = "ERROR: failed to obfuscate"
+            self._log.debug("Failed to obfuscate | err=[%s]", e)
+        return row
+
+    @staticmethod
+    def _sanitize_row(row):
+        # type: (Dict[str]) -> Dict[str]
+        return {key: val for key, val in row.items() if val is not None}
+
+    @staticmethod
+    def _finalize_row(row, statement):
+        # type: (Dict[str], Dict[str]) -> None
+        obfuscated_statement = statement["query"]
+        row["sql_text"] = obfuscated_statement
+        row["query_signature"] = compute_sql_signature(obfuscated_statement)
+
+        metadata = statement["metadata"]
+        row["dd_commands"] = metadata.get("commands", None)
+        row["dd_tables"] = metadata.get("tables", None)
+        row["dd_comments"] = metadata.get("comments", None)
+
+    @staticmethod
+    def _get_estimated_row_size_bytes(row):
+        # type: (Dict[str]) -> int
+        return len(str(row))
+
+    def _create_activity_event(self, active_sessions, active_connections):
+        # type: (List[Dict[str]], List[Dict[str]]) -> Dict[str]
+        return {
+            "host": self._db_hostname,
+            "ddagentversion": datadog_agent.get_version(),
+            "ddsource": "mysql",
+            "dbm_type": "activity",
+            "collection_interval": self.collection_interval,
+            "ddtags": self._tags,
+            "timestamp": time.time() * 1000,
+            "mysql_activity": active_sessions,
+            "mysql_connections": active_connections,
+        }
+
+    @staticmethod
+    def _get_query_from_version(version):
+        # type: (MySQLVersion) -> str
+        # Note, future iterations will have version specific queries
```

Comment:
Would it be better to omit this method entirely until there are version specific queries? It's inviting a potential error when a new version is added and this returns None because it's not in the if statement.

Reply:
Yeah, that's a fair point, we could raise an error if we come across an unsupported version, but we can leave those details out until we need it.